spec/requests/authentication_spec.rb

```diff
@@ -0,0 +1,238 @@
+        post api_v1_user_session_path(email: email, password: password)
+      end
+
+      it 'Should respond with status 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'Should respond with current_user in body' do
+        expect(response.body).to include(email, username)
+      end
+
+      it 'Returns access-token in auth_headers' do
+        expect(response.headers['access-token']).to be_present
+      end
+
+      it 'Returns client in auth_headers' do
+        expect(response.headers['client']).to be_present
+      end
+
+      it 'Returns expiry in auth_headers' do
+        expect(response.headers['expiry']).to be_present
+      end
+
+      it 'Returns uid in auth_headers' do
+        expect(response.headers['uid']).to be_present
+      end
+    end
+
+    context 'When email is invalid' do
+      let(:invalid_email) { "invalid#{email}" }
+      before do
+        post api_v1_user_registration_path(username: username, email: email, password: password)
+        post api_v1_user_session_path(email: invalid_email, password: password)
+      end
+
+      it 'Should respond with status 401' do
+        expect(response).to have_http_status(401)
+      end
+
+      it 'Should respond with error ' do
+        expect(response.body).to include('Invalid login credentials. Please try again')
+      end
+    end
+
+    context 'When password is invalid' do
+      let(:invalid_password) { "invalid#{password}" }
+      before do
+        post api_v1_user_registration_path(username: username, email: email, password: password)
+        post api_v1_user_session_path(email: email, password: invalid_password)
+      end
+
+      it 'Should respond with status 401' do
+        expect(response).to have_http_status(401)
+      end
+
+      it 'Should respond with error ' do
+        expect(response.body).to include('Invalid login credentials. Please try again')
+      end
+    end
+
+    context 'When blunk credentials' do
+      before do
+        post api_v1_user_registration_path(username: username, email: email, password: password)
+        post api_v1_user_session_path(email: '', password: '')
+      end
+
+      it 'Should respond with status 401' do
+        expect(response).to have_http_status(401)
+      end
+
+      it 'Should respond with error ' do
+        expect(response.body).to include('Invalid login credentials. Please try again')
+      end
+    end
+  end
+
+  context 'DELETE api/v1/auth (Destroy user)' do
+    context 'With valid auth_headers' do
+      before do
+        post api_v1_user_registration_path(username: username, email: email, password: password)
+        user = User.last
+        auth_headers = user.create_new_auth_token
+        delete api_v1_user_registration_path, headers: auth_headers
+      end
+
+      it 'Should respond with status 200 with user auth_headers: access-token, client, uid' do
+        expect(response).to have_http_status(200)
+      end
+
```

Comment:
Желательно проверить, что пользователь также исчез и из базы, по аналогии с `Authentication POST api/v1/auth (Sign Up process) With valid data Should create database record`:
```suggestion

it 'Should remove the database record' do
  expect(User.count).to eq(0)
end

```